onnxruntime/core/session/inference_session.cc

```diff
@@ -528,6 +528,11 @@ common::Status InferenceSession::Initialize() {
     // now that all the transforms are done, call Resolve on the main graph. this will recurse into the subgraphs.
     ORT_RETURN_IF_ERROR(graph.Resolve());
 
+    if (!session_options_.optimized_model_filepath.empty()) {
```

Comment:
Right now after L3 optimization the model cannot be serialized... some more work needs to happen in these optimizations before we can do this... 

Please add a check here to make sure optimization level is < 3.  Just FYI this was discussed during design review